### PR TITLE
Feature/id 199/way point implementation

### PR DESCRIPTION
**Pull request by Karvlox**
# **Map Service**

## WHAT?

We implemented the functionality to implement waypoints when performing a search.

## HOW?

Use of MapBox resources to display icons in specific locations by means of test..

## WHY?

So that when a search is performed, the places related to the search are marked on the map.To allow user to watch the map in diferent configurations according to his needs

CODE REVIEWERS:
TEAM A: @Misax148 
TEAM B: @denisgandel
TEAM C: @CatMauCoder

TRAINER
@mmoralesv

Acceptance Criteria

- [ ]  Create a waypoint
- [ ]  Functionality to receive geoJason data from the geographic coordinate and render it as a waypoint on the map.

### CODE REVIEW

File: app/src/main/java/com/isc/hermes/utils/MapConfigure.java [hunk 1]
```diff
@@ -10,12 +12,17 @@
  */
 public class MapConfigure {
 
+    PlacesSearchedManager manager;
```
Comment (Misax148) [inline app/src/main/java/com/isc/hermes/utils/MapConfigure.java:15]: Use encapsulation in your variable please `PlacesSearcherManager manager;`

File: app/src/main/java/com/isc/hermes/utils/MapConfigure.java [hunk 2]
```diff
@@ -10,12 +12,17 @@
...
+    Context context;
```
Comment (Misax148) [inline app/src/main/java/com/isc/hermes/utils/MapConfigure.java:16]: Use encapsulation in your variable please `Context context;`

File: app/src/main/java/com/isc/hermes/utils/PlacesSearchedManager.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import android.content.Context;
+
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+
+/**
+ * This class manages the waypoints on the mapBox map.
+ */
+public class PlacesSearchedManager {
+
+    PlacesSearchedOnTheMap placesSearchedOnTheMap;
```
Comment (Misax148) [inline app/src/main/java/com/isc/hermes/utils/PlacesSearchedManager.java:12]: Use encapsulation in your variable please `PlacesSearchedOnTheMap placesSearchedOnTheMap;`

File: app/src/main/java/com/isc/hermes/utils/PlacesSearchedOnTheMap.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        //pointTest(); //TODO: uncomment this method to test the functionality then this line must be deleted
+        displayPoint(pointList);
+    }
+
+    /**
+     * This method creates and displays waypoints on the map.
+     * @param pointList list of places that will be marked on the map.
+     */
+    public void displayPoint(List<LatLng> pointList) {
+        for (int i = 0; i < pointList.size(); i++) {
```
Comment (Misax148) [inline app/src/main/java/com/isc/hermes/utils/PlacesSearchedOnTheMap.java:35]: Use a **For each** to make it more efficient and clear.

Comment (Karvlox) [reply app/src/main/java/com/isc/hermes/utils/PlacesSearchedOnTheMap.java:35]: Se intento usar un for each, lastimosamente no funcionaba el programa

File: app/src/main/java/com/isc/hermes/utils/MapConfigure.java (hunk 2 above)
Comment (Karvlox) [reply app/src/main/java/com/isc/hermes/utils/MapConfigure.java]: Done.

File: app/src/main/java/com/isc/hermes/utils/MapConfigure.java (hunk 1 above)
Comment (Karvlox) [reply app/src/main/java/com/isc/hermes/utils/MapConfigure.java]: Done.

File: app/src/main/java/com/isc/hermes/utils/PlacesSearchedManager.java (hunk 3 above)
Comment (Karvlox) [reply app/src/main/java/com/isc/hermes/utils/PlacesSearchedManager.java]: Done.

**Review by Misax148 — APPROVED** [review]

**Review by mmoralesv — APPROVED** [review]

File: app/src/main/java/com/isc/hermes/utils/MapConfigure.java [hunk 5]
```diff
@@ -1,5 +1,7 @@
 package com.isc.hermes.utils;
 
+import android.content.Context;
+
```
Comment (CatCoder36) [inline app/src/main/java/com/isc/hermes/utils/MapConfigure.java:4]: As a suggestion, take care of the spaces between imports

**Review by denisgandel — APPROVED** [review]

**Review by CatCoder36 — APPROVED** [review]

**Pull request merged**